antenna/heartbeat.py

```diff
@@ -63,7 +62,9 @@ def heartbeat(self):
         capture unhandled exceptions and report them.
 
         """
-        while self.hb_live:
+        # Keep beating unless the WSGI worker is shutting down AND all
+        # registered lifers are ok with us shutting down
+        while self.is_alive() or any([fun() for fun in _registered_lifers]):
```

Comment:
The `or` here is short-circuiting, so if `is_alive()` returns `True`, then it doesn't call any of the lifers.